dualscreeninfo/android/src/main/kotlin/com/microsoft/reactnativedualscreen/dualscreen/DualScreenInfo.kt

```diff
@@ -113,6 +114,11 @@ class DualScreenInfo constructor(context: ReactApplicationContext) : ReactContex
 		return "landscapeFlipped"
 	}
 
+	private fun convertPixelsToDp(px: Int): Int {
+		val metrics = Resources.getSystem().displayMetrics
+		return (px / (metrics.density)).toInt()
```

Comment:
>(px / (metrics.density)) [](start = 9, length = 24)

We'll need to preserve the fractional components, otherwise the conversion will become lossy.  This function should return a floating point number, and the rects will need to be floats rather than ints.

Reply:
i see that makes more sense. I had that thought but wanted to stay with the convention

Reply:
yah it's a bit unfortunate that we have to make a type change that affects the public api, but I don't know of any customers actually using this property yet (if they did they would have likely noticed the bug you're fixing).  Using floats is required...

---
In reply to: [456238454](https://github.com/microsoft/react-native-dualscreen/pull/46#discussion_r456238454) [](ancestors = 456238454)

Reply:
so it looks like this default map that we are using seems to not support floats. i already set up a meeting with Evgeny for tomorrow morning. i will inquire about the correct solution to this then

Reply:
so the map that we used does not support float, but it does support doubles. i will update the values to a double. and inquire to Evgeny tomorrow morning in our meeting if its possible to change

Reply:
double is fine for this (actually better as the extra precision could help avoid rounding errors further down the pipeline).

---
In reply to: [456260148](https://github.com/microsoft/react-native-dualscreen/pull/46#discussion_r456260148) [](ancestors = 456260148)